backend/tilavarauspalvelu/integrations/opening_hours/hauki_resource_hash_updater.py

```diff
@@ -20,100 +20,73 @@ class HaukiResourceHashUpdater:
-                    continue
-            else:
-                logger.info(f"Updating 'Opening Hours Hash' for resource '{resource['id']}'.")
-
-            # Update the hash and set the latest fetched date to None
-            origin_hauki_resource.opening_hours_hash = resource["date_periods_hash"]
-            origin_hauki_resource.latest_fetched_date = None
-            origin_hauki_resource.save()
-
-            # Delete all new reservable time spans for the resource.
-            # Old time spans are not deleted, as they are kept for archival purposes.
-            origin_hauki_resource.reservable_time_spans.filter(start_datetime__gte=cutoff_date).delete()
-
-            # If there are any ReservableTimeSpans that overlap with the cutoff date, have them end at the cutoff date.
-            # This way we can keep all past data, and have the new data start from the cutoff date.
-            origin_hauki_resource.reservable_time_spans.filter(end_datetime__gte=cutoff_date).update(
-                end_datetime=datetime.datetime.combine(cutoff_date, datetime.time.min, tzinfo=DEFAULT_TIMEZONE)
-            )
-
-            self.resources_updated.append(origin_hauki_resource)
-
-    def _create_reservable_time_spans_for_reservation_units(self) -> int:
-        """Create reservable time spans for all updated OriginHaukiResources."""
-        total_reservable_time_spans_created = 0
-
-        for resource in self.resources_updated:
-            try:
-                client = ReservableTimeSpanClient(resource)
-            except (ReservableTimeSpanClientValueError, ReservableTimeSpanClientNothingToDoError):
-                continue
-            num_created_time_spans = len(client.run())
-            total_reservable_time_spans_created += num_created_time_spans
-            logger.info(f"Created {num_created_time_spans} reservable time spans for resource '{resource}'.")
-
-        if total_reservable_time_spans_created:
-            logger.info(f"Created {total_reservable_time_spans_created} new reservable time spans in total.")
+        if self.total_time_spans_created:
+            logger.info(f"Created {self.total_time_spans_created} new reservable time spans in total.")
         else:
             logger.info("No reservable time spans created.")
-        return total_reservable_time_spans_created
+
+    def _process_single_hauki_resource(self, resource: HaukiAPIResource, *, force_refetch: bool = False) -> None:
+        origin_hauki_resource = OriginHaukiResource.objects.filter(id=resource["id"]).first()
+        if origin_hauki_resource is None:
+            logger.warning(f"OriginHaukiResource with ID '{resource['id']}' was not found.")
+            return
+
+        should_update_resource = (
+            force_refetch
+            or origin_hauki_resource.opening_hours_hash != resource["date_periods_hash"]
+            or origin_hauki_resource.should_update_opening_hours
```

Comment:
I would add an action that takes the `resource["date_periods_hash"]` as an argument, because `origin_hauki_resource.opening_hours_hash != resource["date_periods_hash"]` should always be included in determining if opening hours should be updated.